engine/consensus/dkg/messaging_engine.go

```diff
@@ -104,40 +111,57 @@ func (e *MessagingEngine) Process(_ network.Channel, originID flow.Identifier, e
 func (e *MessagingEngine) process(originID flow.Identifier, event interface{}) error {
 	switch v := event.(type) {
 	case *msg.DKGMessage:
+		// messages are forwarded async rather than sync, because otherwise the message queue
+		// might get full when it's slow to process DKG messages synchronously and impact
+		// block rate.
+		e.forwardInboundMessageAsync(originID, v)
+		return nil
+	default:
+		return fmt.Errorf("invalid event type (%T)", event)
+	}
+}
+
+func (e *MessagingEngine) forwardInboundMessageAsync(originID flow.Identifier, message *msg.DKGMessage) {
+	e.unit.Launch(func() {
 		e.tunnel.SendIn(
 			msg.PrivDKGMessageIn{
-				DKGMessage: *v,
+				DKGMessage: *message,
 				OriginID:   originID,
 			},
 		)
-		return nil
-	default:
-		return fmt.Errorf("invalid event type (%T)", event)
-	}
+	})
 }
 
 func (e *MessagingEngine) forwardOutgoingMessages() {
 	for {
 		select {
 		case msg := <-e.tunnel.MsgChOut:
-			expRetry, err := retry.NewExponential(retryMilliseconds)
-			if err != nil {
-				e.log.Fatal().Err(err).Msg("failed to create retry mechanism")
-			}
+			e.forwardOutboundMessageAsync(msg)
+		case <-e.unit.Quit():
+			return
+		}
+	}
+}
+
+func (e *MessagingEngine) forwardOutboundMessageAsync(message msg.PrivDKGMessageOut) {
+	f := func() {
```

Comment:
```suggestion
	sendMessage := func() {
```
Could use a more descriptive name, or simply pass the function anonymously into `unit.Launch` - eg `unit.Launch(func() { ... })`